public/app/features/dashboard-scene/settings/variables/editors/TextBoxVariableEditor.tsx

```diff
@@ -2,11 +2,22 @@ import React from 'react';
 
 import { TextBoxVariable } from '@grafana/scenes';
 
+import { TextBoxVariableForm } from '../components/TextBoxVariableForm';
+
 interface TextBoxVariableEditorProps {
   variable: TextBoxVariable;
   onChange: (variable: TextBoxVariable) => void;
 }
 
-export function TextBoxVariableEditor(props: TextBoxVariableEditorProps) {
-  return <div>TextBoxVariableEditor</div>;
+export function TextBoxVariableEditor({ variable }: TextBoxVariableEditorProps) {
+  const { value } = variable.useState();
+  const onQueryChange = (event: React.FormEvent<HTMLInputElement>) => {
+    variable.setState({ value: event.currentTarget.value });
+  };
+
+  const onBlur = (event: React.FormEvent<HTMLInputElement>) => {
+    variable.setState({ value: event.currentTarget.value });
+  };
+
+  return <TextBoxVariableForm value={value} onChange={onQueryChange} onBlur={onBlur} />;
```

Comment:
To be reused in different places, I find the props quite confusing... what if we rename the props to be more specific to the fields contained by the form?

Also, we probably don't need both callback. We can just update the state doing onblur, so we save a lot of state changes and variable resolving. WDYT?

```suggestion
  return <TextBoxVariableForm textValue={value} onTextValueBlur={onBlur} />;
```

Reply:
Agree with the `onBlur` comment, that would be sufficient to commit the changes.

No strong opinions on the other comment regarding prop names, don't think this component would be used in other contexts without having it wrapped in the `TextBoxVariableEditor`.

Reply:
🤔 I applied the changes, but I think we forgot something here 🙈 , the `TextBoxVariableForm` is a controlled component, we are using react to handle the `value`  there, in those cases we always need an `onChange` property :)

Reply:
Yep, but then, we should use `setState` in the `onChange`, and `onBlur` would change the scene variable object state instead. 

Setting the state on the component is cheap, and in this case, needed because we want to manage the temporal state as you mentioned. But we could safe a lot of other states updates from the scenes, if we just use `onBlur` for the model change.

 ```jsx
export function TextBoxVariableEditor({ variable }: TextBoxVariableEditorProps) {
  const { value } = variable.useState();
  const {textValue, setTextValueState} = useState(value);
  
  const onQueryChange = (event: React.FormEvent<HTMLInputElement>) => {
    setTextValueState(event.currentTarget.value);
  };

  const onBlur = (event: React.FormEvent<HTMLInputElement>) => {
    variable.setState({ value: textValue });
  };

  return <TextBoxVariableForm value={textValue} onChange={onQueryChange} onBlur={onBlur} />;
```

Reply:
Those are great suggestions :), applying them 👩🏾‍💻

Reply:
@ivanortegaalba I've made the changes 😄